### ISSUE TITLE
Exit with exit code 1 if coverage threshold is not met

### ISSUE THREAD

**Issue by gajus** [rpl/flow-coverage-report#18]
This is useful for `precommit` checks and CI integration.

**Comment by rpl** [comment]
@gajus it should currently exit with an exit code 2 (https://github.com/rpl/flow-coverage-report/blob/master/src/lib/cli/index.js#L25), but probably it would be better to use an exit 1 (I can't actually remember if there is a real reason why I'm exiting with the exit code 2 instead of 1).

In your tests, does it exit with code 2 or it is not exiting with an error exit code at all?

**Comment by alebiavati** [comment]
I'm getting the same problem, but I don't think it matters wether it exits with code 1 or 2.

The problem that I am seeing after diving into the code a little is that the `waitForDrain` promise in the text report generator never resolves, causing the threshold check to never happen.

When running the following command the script does NOT throw an error:
```
flow-coverage-report -i './src/**/*.js' -t html -t text --threshold 90
```

But if I run the same command without the text report the script exits correctly (with error code 2):
```
flow-coverage-report -i './src/**/*.js' -t html --threshold 90
```

(for the test to work you have to have some non-typed JS in the `src` folder)

It seems that `process.stdout.on('drain', ...)` never triggers and therefor the promise never resolves. I don't know how to solve that yet, but I thought I'd share.

**Comment by rpl** [comment]
@gajus Thanks again for reporting this issue, and thanks to @alebiavati for the analysis, I've just merged #22 which should fix this issue.

The fix has been released on npm as flow-coverage-report v0.3.0, let me know if the merged fix is fixing your issue as expected.
